### DISCUSSION TITLE
Heltec-ESP32 V3 Position Nonsense

### DISCUSSION THREAD

**Discussion by ikearat** [meshtastic/Discussions#19] [ESP32]
I briefly looked into the hole before I jumped in and...
Landed on a Heltec-ESP32 V3 "turn key" setup from 3Dsafe on Etsy.
It's working just fine with one irritating issue. It keeps "moving" around. I have the position precision set to max and am using a manual set of coordinates but after a few moments, the damned thing ghosts back to some position about 2 miles away from where I am setting the location. I read as much as I could find on the old Discorse site and while most of the information is "old" ( like over two years old, ancient firmware versions in use ) I could find nothing definite to help me resolve this nonsense.
If anyone wants to chip in with a clue, I'm down here in my Meshtastic hole looking up at the bright spot.
System details:
Android app Version: 2.5.3 ( not passing GPS data to the Meshtastic app. I'm indoors and the phone is satellite blind. )
Android OS Version: 11 on a Blackview BL8800Pro ( Netguard firewall )
Heltec-ESP32 Version: Heltec-ESP32 V3 Firmware 2.5.6
For build details please visit the related Etsy user page.

edited for spelling...

**Comment by adingbatponder** [comment]
Are you using fixed position? If so, is it that fixed position setting you are losing?
![Screenshot_20241029-203557~2](https://github.com/user-attachments/assets/e9201939-c721-475a-b9b6-22ee41c141fc)

**Comment by ikearat** [comment]
Yes. I set the Fixed Position with the slider to the right.
I did a bunch of experimenting with no "fix" in the results.
I turned off all of the location settings in both the Heltec and the Android app. As expected the node vanished from the map on the Android app. I did not try this with MQTT enabled, but when I was using MQTT I had the same location issues as now.
I tried turning on "fixed position" and allowing the node to reach into it's own drawers and "find it's self". Weirdly, the location is chooses is the same position, about 2 miles away from it's physical location... which it should have no idea what that is, the node does not have a GPS board and the phone location is turned off. Why it chooses this location instead of defaulting to someplace elsewhere in the World is part of the nonsense.
I tried using DMS instead of decimal location, which seems to only alter the published location and not change the input field in Position settings. I screwed around with the slider for location precision. All that seems to do is provide a guesstimate to the mapping aspect to control the circle around the location.
I have tried everything I could find on the Heltec product website, and everything (relative to firmware version) on the Discorse site and even asked my brother who is a paid software engineer to look at the code. He declined due to the instant revulsion with the "horrible network protocol" :)
This whole exercise is nonsense. I am in strong suspect that the Meshtastic program running in the Heltec is the problem but I have no path to correcting this. Whomever created this code will have to answer or all of this becomes a money pit. I will spend my next Meshtastic dollar on bicycle parts.

**Reply by adingbatponder** [reply]
The only thing I can offer is confirming that something like that happened on my Station G2 which systematically changed the set fixed position to another down the road. On my local Mesh whatsapp one person said that the app has a bug and the positions can change when the channels are changed, or if you change preset values somewhere, and also recomended not using more than 4 numbers after the decimal place. That is not a solution but may point to the crazy interdependence of the settings.

**Reply by ikearat** [reply]
While you were typing your reply I was fooling around with the number of digits I supplied after the decimal. I thought I was making headway when it all went to hell again. If I recall, I had offered a 6th digit at the time. I will go back and see if supplying only 5 digits is stable.

**Reply by ikearat** [reply]
nope. Still resets to the same location. Were in the software is this stuff getting the information is is substituting for user input and why is it ignoring the "Use Fixed Position" setting... this acts like android when you are trying to set other crap in the system. I know it's not the android app, it has to be the Mestastic program on the Heltec. Why, what, what the heck?!? I'm going to recharge the battery on this thing and then turn it off for a week and see if this new and easier to keep developers and users in sync forum comes up with some stability. Why is the damned reply box here not expanding to encompass the entire posting? It's acting like a text messaging app. Stupid cell phones even messing with webpage designers at this level. Yes, frustrated.

**Reply by adingbatponder** [reply]
What shocks me is that we are in this horrible github thing to help the developers organise but none seem to answer. There is a vast amount of frustration brewing about everything.

**Reply by Ed-Mouse** [reply]
I had been having this issue with the last few firmware releases on my Heltec V3.

I set the position so that the Android app would plot my location but I did not want to share the position over the mesh.

When I set up some devices back in January, the position was always shared with channel 0 so I created a private channel on 0 for chatting with family, and put the LongFast channel on 1.

The fixed position stopped changing for me when I went to Radio Configuration, Channels then selected my private Channel on 0, and set 'Position enabled' and 'Precise location' to on.

Things have changed since I set up my channels and now you can share position information on all channels so you can enable Precise location on any private channel and maybe that will stop your position changing. Worth a try.

**Reply by ikearat** [reply]
Ed-Mouse,
Thank you. My node has remained where I told it to be for 3+ minutes now. Longest time on record yet!
I am about to try sending a message to one of the nodes I can "see" from where I am. (4 minutes now...)
Only after revisiting your post did I realize the the rodent thing :)
Later I will pretend to understand why what you suggested is working (6 minutes...) Give me time, I have only had this thing for a few days. I'm still a little miffed at the information flow but if there was a way to "upvote" or endorse you as a support source, you would get the star :)
8 minutes...

Thank you!
ikearat

**Comment by ikearat** [comment]
Follow-up: I screwed around with channel management (or in my case mismanagement) and ended up with a single Long Fast channel (channel 0 [zero]) which has held the fixed coordinates for several hours. I am "seeing" different nodes than before, one is a "router" and I think it is hosting two other nodes which show as "unlocked", are two hops away and don't have GPS coordinates (also don't show on the map).
I created the (my) channel by deleting channel 1 (which I had made using Ed-Mouse's suggestion, thank you again) and then "reset" channel 0. I then named channel 0 to "LongFast" and confirmed the frequency slot to be 20 in line with use in the US. The one thing that has me a little bothered is the PSK I am using. I am using "AQ==" for PSK. I did try a resource from the Meshtastic firmware stuff on Github showing a bunch of QR code images. I fooled around with the one listed for the US and don't know if that was effective for anything. I would like to arrive at a setup that exposes the original nodes I was seeing when I first turned on my node. I will monitor a few days and see if they are just off-line or have changed their node names. Meshtastic would benefit with some entry level user guides. Sure, the documentation is rich and covers a lot of stuff but I bet you some money Meshtastic has lost a lot of "customers" and possibly chased away some talented programmers for the lack of clear easy to understand startup docs.
This edit box is still a waste of screen space.

**Reply by adingbatponder** [reply]
You hit the nail on the head. A "how to" or "FAQ" resource would help massively. The settings required for certain things can be non-intuitive. I cannot find a guide on how to get Store & Forward working on 2.5 ... but think I know, but want confirmation before I "claim to have a step by step solution" to share. I do not want the easy way out, I just want informative dialogue. So far here this answer above concerning the channels has been the only real help on any topic. I will try it. Thanks for persisting and sharing.

**Comment by ikearat** [comment]
"It's baaaak"
Nonsense returns. The thing will hold a LAT / LON but refuses to hold an altitude. AND, you have to enter the manual data in meters even while you set the display units to imperial. How about just using the system defaults for data format? That-a-ways it would be in the correct format for the user without having to hop around doing metric <--> imperial conversions and the other way 'round.
Also, when you are using the fields for entering LAT / LON and altitude, and you live in a part of the World where your Longitude numeric is proceeded with a (-) like -95.0000 the popup keyboard does not offer a (-) key to press. You have to go elsewhere and use a text input field and copy 'n paste the Longitude. That's not the end of it... when you tap in the field to make our changes, the "0.0" is not replaced with your input or sometimes it is. Inconsistent. Ok. I have learned from this experience, don't fool around with your Meshtastic junk late at night. Go sleep instead.
